pkgs/by-name/rm/rmenu/package.nix

```
@@ -0,0 +1,82 @@
+{ fetchFromGitHub
+, glib
+, gtk3
+, lib
+, libsoup_3
+, networkmanager
+, pkg-config
+, rustPlatform
+, webkitgtk_4_1
+, wrapGAppsHook
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "rmenu";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    rev = "v${version}";
+    owner = "imgurbot12";
+    repo = "rmenu";
+    hash = "sha256-mzY+M7GGJDxb8s7pusRDo/xfKE/S4uxPy4klRBjVGOA=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    glib
+    gtk3
+    libsoup_3
+    networkmanager
+    webkitgtk_4_1
+  ];
+
+  strictDeps = true;
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "gio-0.19.0" = "sha256-+PAQNJ9sTk8aKAhA/PLQWDCKDT/cQ+ukdbem7g1J+pU=";
+      "nm-0.4.0" = "sha256-53ipJU10ZhIKIF7PCw5Eo/e/reUK0qpyTyE7uIrCD88=";
+    };
+  };
+
+  postInstall = ''
+    # copy themes and plugins
+    mkdir $out/themes
+    mkdir $out/plugins
+    cp -vfr $src/themes/* $out/themes/.
+    cp -vfr $src/other-plugins/* $out/plugins/.
+    mv $out/bin/* $out/plugins # everything is a plugin by default
+
+    # rmenu and rmenu-build are actual binaries
+    mv $out/plugins/rmenu $out/bin/rmenu
+    mv $out/plugins/rmenu-build $out/bin/rmenu-build
+
+    # fix plugin names
+    # desktop  network  pactl-audio.sh  powermenu.sh  run  window
+    mv $out/plugins/run $out/plugins/rmenu-run
+    mv $out/plugins/desktop $out/plugins/rmenu-desktop
+    mv $out/plugins/network $out/plugins/rmenu-network
+    mv $out/plugins/window $out/plugins/rmenu-window
+
+    # fix config and theme
+    mkdir -p $out/etc/xdg/rmenu
+    cp -vf $src/rmenu/public/config.yaml $out/etc/xdg/rmenu/config.yaml
+    sed -i "s@~\/\.config\/rmenu\/themes@$out\/themes@g" $out/etc/xdg/rmenu/config.yaml
+    sed -i "s@~\/\.config\/rmenu@$out\/plugins@g" $out/etc/xdg/rmenu/config.yaml
+    ln -sf  $out/themes/dark.css $out/etc/xdg/rmenu/style.css
+  '';
+
```

```suggestion
preFixup = ''
  gappsWrapperArgs+=(
    --suffix XDG_CONFIG_DIRS : "$out/etc/xdg"
  )
'';
```

Otherwise it won't use your shipped config file.

(Test it yourself first)

---

Good catch, i didn't know this was a thing but makes sense for profiles and shells. Tested and works great :)
With this change, does it still make sense to use `$out/etc/xdg` which will be auto-linked into `run/current-system/sw/etc/xdg/`, or would a different (not linked) path make more sense how that the wrapper manages the XDG_CONFIG_DIRS that are set for it?

---

Alright, tested the approach in share and works just as well. Considerign this might indeed be cleaner i'll keep it in share. Thank you again for the constructive review.